scene/2d/gpu_particles_2d.cpp

```diff
@@ -173,6 +175,10 @@ void GPUParticles2D::set_process_material(const Ref<Material> &p_material) {
 	RS::get_singleton()->particles_set_process_material(particles, material_rid);
 
 	update_configuration_warnings();
+
+	if (Engine::get_singleton()->is_editor_hint()) {
```

Comment:
This condition is not needed. The signal doesn't do anything outside editor.

Reply:
Yes, the signal is emitted, but nothing connects to it at runtime (unless a user does it for whatever reason). For that reason, the editor hint check is unnecessary and it's not done anywhere (AFAIK).